apps/server/src/modules/registration/api/registration.uc.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@nestjs/common';
+import { Permission } from '@shared/domain/interface';
+import { EntityId } from '@shared/domain/types';
+import { CreateRegistrationBodyParams } from './dto/request/create-registration.body.params';
+import { Registration, RegistrationService } from '../domain';
+import { Action, AuthorizationService } from '@modules/authorization';
+import { RegistrationFeatureService } from './service';
+import { RoomService } from '@modules/room';
+import { RoomPermissionService } from '@modules/room/api/service';
+
+@Injectable()
+export class RegistrationUc {
+	constructor(
+		private readonly authorizationService: AuthorizationService,
+		private readonly registrationService: RegistrationService,
+		private readonly registrationFeatureService: RegistrationFeatureService,
+		private readonly roomService: RoomService,
+		private readonly roomPermissionService: RoomPermissionService
+	) {}
+
+	public async createRegistration(userId: EntityId, props: CreateRegistrationBodyParams): Promise<Registration> {
+		this.registrationFeatureService.checkFeatureRegistrationEnabled();
+
+		const user = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkOneOfPermissions(user, [Permission.USER_CREATE]);
+
+		const existingRegistration = await this.fetchAndUpdateRegistrationByEmail(props.email, props.roomIds);
+		if (existingRegistration) {
+			return existingRegistration;
+		}
+
+		const registration = await this.registrationService.createRegistration({ ...props });
+		return registration;
+	}
+
+	public async getSingleRegistrationByHash(registrationHash: string): Promise<Registration> {
+		this.registrationFeatureService.checkFeatureRegistrationEnabled();
+
+		const registration = await this.registrationService.getSingleRegistrationByHash(registrationHash);
+
+		return registration;
+	}
+
+	public async getRegistrationsByRoomId(userId: EntityId, roomId: EntityId): Promise<Registration[]> {
+		this.registrationFeatureService.checkFeatureRegistrationEnabled();
+
+		await this.roomService.getSingleRoom(roomId);
+		await this.roomPermissionService.checkRoomAuthorizationByIds(userId, roomId, Action.write);
```

Comment:
lets avoid making things public that arent already. I would instead do
```
const roomMembershipAuthorizable = await this.roomMembershipService.getRoomMembershipAuthorizable(roomId);
this.authorizationService.checkPermission(user, roomMembershipAuthorizable,  AuthorizationContextBuilder.write([]));
```

We should also add this check for createRegistration.